src/Equinor.ProCoSys.IPO.Command/InvitationCommands/UpdateRfocAcceptedStatus/UpdateRfocAcceptedCommandHandler.cs

```diff
@@ -0,0 +1,89 @@
+{
+    public class UpdateRfocAcceptedCommandHandler : IRequestHandler<UpdateRfocAcceptedCommand, Result<Unit>>
+    {
+        private readonly IInvitationRepository _invitationRepository;
+        private readonly IProjectRepository _projectRepository;
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IPlantProvider _plantProvider;
+        private readonly ICertificateApiService _certificateApiService;
+        private readonly ILogger<UpdateRfocAcceptedCommandHandler> _logger;
+
+        public UpdateRfocAcceptedCommandHandler(
+            IInvitationRepository invitationRepository,
+            IProjectRepository projectRepository,
+            IUnitOfWork unitOfWork,
+            IPlantProvider plantProvider,
+            ICertificateApiService certificateApiService,
+            ILogger<UpdateRfocAcceptedCommandHandler> logger)
+        {
+            _invitationRepository = invitationRepository;
+            _projectRepository = projectRepository;
+            _unitOfWork = unitOfWork;
+            _plantProvider = plantProvider;
+            _certificateApiService = certificateApiService;
+            _logger = logger;
+        }
+
+        public async Task<Result<Unit>> Handle(UpdateRfocAcceptedCommand request, CancellationToken cancellationToken)
+        {
+            var project = await _projectRepository.GetProjectOnlyByNameAsync(request.ProjectName);
+            if (project == null)
+            {
+                _logger.LogInformation($"Early exit in RfocAccepted handling. Project {request.ProjectName} does not exists in IPO module");
+                return new SuccessResult<Unit>(Unit.Value);
+            }
+            
+            if (project.IsClosed)
+            {
+                _logger.LogInformation($"Early exit in RfocAccepted handling. Project {request.ProjectName} is closed in IPO module");
+                return new SuccessResult<Unit>(Unit.Value);
+            }
+
+            var certificateMcPkgsModel = await _certificateApiService.TryGetCertificateMcPkgsAsync(_plantProvider.Plant, request.ProCoSysGuid);
+
+            if (certificateMcPkgsModel == null)
+            {
+                var error = $"Certificate {request.ProCoSysGuid} McPkg scope not found";
+                _logger.LogError(error);
+                return new NotFoundResult<Unit>(error);
+            }
+
+            var certificateCommPkgsModel = await _certificateApiService.TryGetCertificateCommPkgsAsync(_plantProvider.Plant, request.ProCoSysGuid);
+
+            if (certificateCommPkgsModel == null)
+            {
+                var error = $"Certificate {request.ProCoSysGuid} CommPkg scope not found";
+                _logger.LogError(error);
+                return new NotFoundResult<Unit>(error);
+            }
+
+            if (!certificateMcPkgsModel.CertificateIsAccepted && !certificateCommPkgsModel.CertificateIsAccepted)
```

Comment:
I don't think that should ever happen. I think we can have CommPkg != accepted && McPkg != accepted (in which case there is nothing to update), and we can get CommPkg != accepted but McPkg == accepted

But you are suggesting adding a check for the case you are suggesting and logging that there is something wrong?

Reply:
We're only doing early return in the case both are accepted, if thats correct, then A-OK. Otherwise we need to make sure our logic catches all eventualities.